solutions/devsprint-bruno-ramos-2/DeliveryAppChallenge/Modules/Product/RestaurantDetails/RestaurantDetailsInteractor.swift

```diff
@@ -36,10 +36,10 @@ extension RestaurantDetailsInteractor: RestaurantDetailsBusinessLogic {
             guard let self = self else { return }
 
             switch result {
-            case .success(let name):
-                self.presenter.presentRestaurantMenu(response: .success(name))
-            case .failure(let error):
-                self.presenter.presentRestaurantMenu(response: .failure(RestaurantFetchDataError()))
+            case .success(let restaurantDetails):
+                self.presenter.presentRestaurantMenu(response: .success(restaurantDetails))
+            case .failure(_):
+                self.presenter.presentRestaurantMenu(response: .failure(APIError.networkError))
```

Comment:
Nesse caso, o ideal é que a própria camada de network fique responsável por definir qual é o tipo de erro vindo da API, então nesse caso, construiríamos um erro aqui, somente se precisássemos mapear algum erro definido por regra de negócio. Então o ideal é que só seja passado o erro vindo do worker, juntamente com a camada de network
```suggestion
            case .failure(let error):
                self.presenter.presentRestaurantMenu(response: .failure(error))
```

Reply:
Feito :white_check_mark: